lib/header.c

```diff
@@ -568,15 +568,15 @@ static int regionSwab(indexEntry entry, int il, int dl,
 	    }
 	}   break;
 	case RPM_INT32_TYPE:
-	{   int32_t * it = ie.data;
+	{   uint32_t * it = ie.data;
```

Comment:
See the INT64 type handling above.  Additionally `htonll()`, `htonl()`, and `htons()` all work with unsigned integers so the casting as signed integers, while probably innocuous, is confusing.